packages/suite/src/components/suite/DeviceDisplay.tsx

```diff
@@ -0,0 +1,209 @@
+    const addressDisplayType = useSelector(selectAddressDisplayType);
+
+    const areChunksUsed =
+        addressDisplayType === AddressDisplayOptions.CHUNKED && !unavailableCapabilities?.chunkify;
+    const isPixelType = selectedDeviceInternalModel !== DeviceModelInternal.T2T1;
+    const isPaginated = network === 'cardano';
+    const iconSize = isPixelType ? 10 : 20;
+    const iconColor = isPixelType ? '#ffffff' : '#959596';
+    const iconNextName = isPixelType ? 'ADDRESS_PIXEL_NEXT' : 'ADDRESS_NEXT';
+    const iconContinuesName = isPixelType ? 'ADDRESS_PIXEL_CONTINUES' : 'ADDRESS_CONTINUES';
+
+    const showChunksInRows = (chunks: ReactNode[][] | undefined, isNextAddress?: boolean) =>
+        chunks?.map((row, rowIndex) => (
+            <Row
+                // eslint-disable-next-line react/no-array-index-key
+                key={rowIndex}
+                isPixelType={isPixelType}
+                isAlignedRight={rowIndex === 0 && isNextAddress}
+            >
+                {row.map((chunk, chunkIndex) => (
+                    // eslint-disable-next-line react/no-array-index-key
+                    <Text isPixelType={isPixelType} key={chunkIndex}>
+                        {chunk}
+                    </Text>
+                ))}
+            </Row>
+        ));
+
+    const renderChunks = (address: string) => {
+        const chunks = address.match(/.{1,4}/g);
+
+        // Add pagination icons
+        // @ts-expect-error
+        chunks?.splice(15, 0, <Icon size={iconSize} icon={iconNextName} color={iconColor} />);
+        // @ts-expect-error
+        chunks?.splice(16, 0, <Icon size={iconSize} icon={iconContinuesName} color={iconColor} />);
+
+        const groupedChunks = chunks?.reduce((result, chunk, index) => {
+            const rowIndex = Math.floor(index / 4);
+
+            if (!result[rowIndex]) {
+                result[rowIndex] = [];
+            }
+
+            result[rowIndex].push(chunk);
+
+            return result;
+        }, [] as ReactNode[][]);
+
+        if (isPaginated) {
+            const PAGE_SIZE = 4;
+            return (
+                <Content>
+                    <Chunks isPixelType={isPixelType}>
+                        {showChunksInRows(groupedChunks?.slice(0, PAGE_SIZE))}
+                    </Chunks>
+                    <Wrapper>
+                        <Divider isPixelType={isPixelType} />
+                        <AddressLabel isPixelType={isPixelType}>
+                            <Translation id="NEXT_PAGE" />
+                        </AddressLabel>
+                    </Wrapper>
+                    <Chunks isPixelType={isPixelType}>
+                        {showChunksInRows(groupedChunks?.slice(PAGE_SIZE), true)}
+                    </Chunks>
+                </Content>
+            );
+        }
+
+        return <Chunks isPixelType={isPixelType}>{showChunksInRows(groupedChunks)}</Chunks>;
+    };
+
+    const renderOriginal = (address: string) => {
```

Comment:
I see, the icons are hadrer to handle without `absolute` in the continuous version. Not saying you have to reqrite, but since you are always slicing the same amounts of characters you can slice them into rows as well and use the grid approach, just with a different grid, then you won't need absolute positioning and pixel peep ;)

Reply:
Let's keep it like it is, it works and doesn't look that bad